### PR TITLE
#3 Some Fixes and Features

### PR DESCRIPTION
**Pull request by Romazes**
##### Feature
1. Added `[Explicit]` attribute in tests, because we need setup config. TDA doesn't have public endpoint.
2. Created independent file to test `HistoryProvider` brokerage.
3. Cleared code in main brokerage test file (`TDAmeritradeBrokerageTests.cs`). 
4. Lock object in PlaceOrder to sync threads.
###### Fix
1. Price fildes for MarketStop and StopLimit orderType.
2. assing response in [Execute method](https://github.com/QuantConnect/Lean.Brokerages.TDAmeritrade/compare/master...Romazes:Lean.Brokerages.TDAmeritrade:develop?expand=1#diff-d4ceb49207b3088f6c19d04ac582c16667e95b3e2642379d479421261e0c0896R119)
3. Update order: After we updated order in Lean then we have removed order from cached collection.
4. `CancelOrder()` method
5. `GetOpenOrders()`. Also, open orders add to inner cache collection to get response from websocket

#### Related Issue

N/A

#### Motivation and Context

TDA implementation.

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

Run test locally use your config.
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs [hunk 1]
```diff
@@ -134,7 +138,7 @@ private T Execute<T>(RestRequest request)
                     return (T)(object)untypedResponse.Content;
                 }
 
-                return JsonConvert.DeserializeObject<T>(untypedResponse.Content);
+                response = JsonConvert.DeserializeObject<T>(untypedResponse.Content);
```
Comment (borseno) [inline QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs:141]: error here

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs [hunk 2]
```diff
@@ -284,15 +251,19 @@ public override List<Holding> GetAccountHoldings()
 
             var holdings = new List<Holding>(positions.Count);
 
+            var quotes = GetQuotesLastPrice(positions.Select(x => x.ProjectedBalances.Symbol));
+
             foreach (var hold in positions)
             {
-                var symbol = Symbol.Create(hold.ProjectedBalances.Symbol, SecurityType.Equity, Market.USA);
+                var brokerageSymbol = hold.ProjectedBalances.Symbol;
```
Comment (Martin-Molinero) [inline QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs:258]: Should use the SymbolMapper to convert from brokerage symbol into lean symbol 👌

Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs:258]: Implemented `GetLeanSymbol()`

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs [hunk 3]
```diff
@@ -161,15 +165,18 @@ public override bool PlaceOrder(Order order)
                 return false;
             }
 
-            // After we have gotten websocket, we will dequeue order from queue
-            _submitedOrders.TryDequeue(out var orderResponse);
+            lock(_onPLaceOrderLockObject)
```
[the remark this answers is not in the document]
Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.cs]: Fixed.

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**